lib/foreman_tasks_core/runner/parent.rb

```diff
@@ -10,21 +10,23 @@ def initialize(targets = {}, suspended_action: nil)
       end
 
       def generate_updates
-        @outputs.reduce({}) do |acc, (key, value)|
+        base = {}
+        base[@suspended_action] = Runner::Update.new(ForemanTasksCore::ContinuousOutput.new, @exit_status) if @exit_status
+        @outputs.reduce(base) do |acc, (key, value)|
           if value.empty? && @exit_status.nil?
             acc
           else
             identifier = key
             @outputs[identifier] = ForemanTasksCore::ContinuousOutput.new
-            key = host_action(identifier) unless identifier == @suspended_action
+            key = host_action(identifier)
             exit_status = @exit_statuses[identifier] || @exit_status if @exit_status
             acc.merge(key => Runner::Update.new(value, exit_status))
```

Comment:
could we rename the variable? it is confusing to redefine the key from `hostname<String>` to `Action::Suspended` I believe that identifier is now useless assignment and we could have:
```suggestion
            identifier = host_action(key)
            exit_status = @exit_statuses[identifier] || @exit_status if @exit_status
            acc.merge(identifier => Runner::Update.new(value, exit_status))
```

Or maybe the other way around `key -> identifier` being the hostname directly comming to the loop block and keeping the key as the calculated key as is